### PR TITLE
#8921 Fix content script injection on sandboxed srcdoc iframes

### LINKED ISSUE #8921: Bugfix: workaround to inject content script on iframes with `srcdoc` and `sandbox` attributes
A bug was introduced in chromium and included in the latest version of Chrome 127 that prevents the content script from loading in iframes with both the `srcdoc` and `sandbox` attributes set, e.g. `<iframe srcdoc="..." sandbox/>`. See slack channel [#bugfix-sandbox-iframes-chrome-update](https://pixiebrix.slack.com/archives/C07DQ2J7C78) for more details.

The proposed workaround is programmatically removing the sandbox attribute and reloading the iframe to trigger content script injection. This [strategy was discussed in the past](https://github.com/pixiebrix/pixiebrix-extension/issues/7300#issuecomment-1882418119) regarding issues with sandboxed iframes, but a different approach was taken at the time.

**Implementation sketch**
- Add some logic to content script initialization that will iterate over all iframes on the page
- Remove the sandbox attr, if present, and reload the frame

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #8921

### CODE REVIEW

File: src/contentScript/contentScriptCore.ts [hunk 1]
```diff
@@ -81,6 +81,14 @@ export async function init(): Promise<void> {
   initRuntime(brickRegistry);
   initDeferredLoginController();
 
+  for (const iframe of document.querySelectorAll("iframe")) {
+    if (iframe.hasAttribute("sandbox")) {
+      iframe.removeAttribute("sandbox");
+      // eslint-disable-next-line no-self-assign -- force the iframe to reload
```
Comment (fungairino) [inline src/contentScript/contentScriptCore.ts:87]: Can we reference and link to the chrome issue here?
https://issues.chromium.org/issues/355256366

**Review by fungairino — APPROVED** [review]

File: src/contentScript/contentScriptCore.ts [hunk 2]
```diff
@@ -81,6 +81,14 @@ export async function init(): Promise<void> {
   initRuntime(brickRegistry);
   initDeferredLoginController();
 
+  for (const iframe of document.querySelectorAll("iframe")) {
+    if (iframe.hasAttribute("sandbox")) {
```
Comment (fungairino) [inline src/contentScript/contentScriptCore.ts:85]: as mentioned in slack huddle, we should check for the srcdoc attribute as well

File: src/contentScript/contentScriptCore.ts (hunk 1 above)
Comment (mnholtz) [reply src/contentScript/contentScriptCore.ts]: Great idea!

**Review by fungairino — APPROVED** [review]

**Comment by fregante** [comment]
Have you tried enabling `match_origin_as_fallback` instead? This might fall into the "frame that uses data:" set

**Comment by twschiller** [comment]
@fungairino @mnholtz see https://github.com/pixiebrix/pixiebrix-extension/pull/8922#issuecomment-2249352218

**Comment by mnholtz** [comment]
> Have you tried enabling match_origin_as_fallback instead? This might fall into the "frame that uses data:" set

See eduardo's comment here: https://github.com/pixiebrix/pixiebrix-extension/issues/8921#issuecomment-2251056110
We were both unable to produce a fix by setting `match_origin_as_fallback` with our project specifically. @fungairino did mention that he was able to replicate a fix with the simple reproduction included in https://developer.chrome.com/docs/extensions/develop/concepts/content-scripts

We will tentatively open a follow-up ticket to investigate, but seeing that we already have a working solution, I'm not sure what the priority will be.

**Pull request merged**